jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/RFC6265SetCookieParser.java

```diff
@@ -0,0 +1,186 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+/**
+ * <p>A parser for {@code Set-Cookie} header values following
+ * <a href="https://datatracker.ietf.org/doc/html/rfc6265">RFC 6265</a>.</p>
+ * <p>White spaces around cookie name and value, and around attribute
+ * name and value, are permitted but stripped.
+ * Cookie values and attribute values may be quoted with double-quotes.</p>
+ */
+public class RFC6265SetCookieParser implements SetCookieParser
+{
+    @Override
+    public HttpCookie parse(String setCookieValue)
+    {
+        // Implementation of the algorithm from RFC 6265, section 5.2.
+
+        // Parser state.
+        State state = State.NAME;
+        String name = null;
+        boolean quoted = false;
+        HttpCookie.Builder cookie = null;
+        int offset = 0;
+        int length = setCookieValue.length();
+
+        // Parse.
+        for (int i = 0; i < length; ++i)
+        {
+            char ch = setCookieValue.charAt(i);
+            switch (state)
+            {
+                case NAME ->
+                {
+                    HttpTokens.Token token = HttpTokens.getToken(ch);
+                    if (token == null)
+                        return null;
+                    if (ch == '=')
+                    {
+                        name = setCookieValue.substring(offset, i).trim();
+                        if (name.isEmpty())
+                            return null;
+                        offset = i + 1;
+                        state = State.VALUE_START;
+                    }
+                }
+                case VALUE_START ->
+                {
+                    if (Character.isWhitespace(ch))
```

Comment:
Would checkout for Token SPACE and HTAB be better to precisely match the spec, as currently this is true for line feeds, vertical tabs etc.

Reply:
The annoyance of using tokens is that you have to check for `null` before testing the type.
I'm also forgiving on chars for names and values where I don't call `isRfc6265CookieOctet()`.

Reply:
comment that it is more forgiving.